### PR TITLE
makefiles: expose a single DEFAULT_CHANNEL variable

### PR DESCRIPTION
**Pull request by aabadie**
The idea of this PR is to use a single variable for setting the default channel when building a firmware. It can be use like this:
```
$ make BOARD=samr21-xpro DEFAULT_CHANNEL=23 flash term
```

### CODE REVIEW

File: drivers/include/at86rf2xx.h [hunk 1]
```diff
@@ -63,12 +63,18 @@ extern "C" {
 /* the AT86RF212B has a sub-1GHz radio */
 #define AT86RF2XX_MIN_CHANNEL           (0)
 #define AT86RF2XX_MAX_CHANNEL           (10)
+#ifdef DEFAULT_CHANNEL
+#define AT86RF2XX_DEFAULT_CHANNEL (DEFAULT_CHANNELU)
```
Comment (Kijewski) [inline drivers/include/at86rf2xx.h]: DEFAULT_CHANNEL***U***? Typo?

Comment (aabadie) [inline drivers/include/at86rf2xx.h]: Well, no, just a misunderstanding of macro on my side... I removed the added _U_'s. I wanted to keep the initial `(26U)` notation using the defined variable. Maybe using `##` could make it but seems a bit too complex.

**Comment by OlegHahm** [comment]
ACK

**Comment by OlegHahm** [comment]
needs squashing

**Comment by aabadie** [comment]
squashed, let's wait for Murdock

**Comment by kYc0o** [comment]
Please include the rest of the drivers, if the intention is to allow this functionality to any radio. Otherwise, why only this radios are selected?

**Comment by aabadie** [comment]
> Otherwise, why only these radios are selected?

Because there are no other drivers supporting this functionality in master. Only CC110X could be a candidate but it's not a 802.15.4 transceiver.

**Comment by aabadie** [comment]
squashed

**Comment by kYc0o** [comment]
So no support for other transceivers?

**Comment by aabadie** [comment]
> So no support for other transceivers?

do you mean cc2420 ? this is the only one I see.

**Comment by kYc0o** [comment]
and cc2538 ;)

**Comment by aabadie** [comment]
@kYc0o, I added the support for both cc2420 and cc2538

File: cpu/cc2538/include/cc2538_rf.h [hunk 2]
```diff
@@ -58,7 +58,12 @@ extern "C" {
 #define CC2538_MAX_FREQ             (2507)
 
 #define CC2538_RF_POWER_DEFAULT     (3)    /**< Default output power in dBm */
+#ifdef DEFAULT_CHANNEL
+#define CC2538_RF_CHANNEL_DEFAULT   (DEFAULT_CHANNEL)
+#endif
+#ifndef CC2538_RF_CHANNEL_DEFAULT
 #define CC2538_RF_CHANNEL_DEFAULT   (26U)
+#endif
```
Comment (kYc0o) [inline cpu/cc2538/include/cc2538_rf.h]: Why not a simple `#else` ?

Comment (aabadie) [inline cpu/cc2538/include/cc2538_rf.h]: @kYc0o, I wanted the same behaviour as the one used in the other transceivers. Developers are free to use the global `DEFAULT_CHANNEL` variable or the transceiver specific one (`CC2538_RF_CHANNEL_DEFAULT` here).

Comment (kYc0o) [inline cpu/cc2538/include/cc2538_rf.h]: Ok great, let's continue with this.

**Comment by kYc0o** [comment]
all green + ACK -> GO!

**Pull request merged**